### PR TITLE
Add a two-stage review process.

### PR DESCRIPTION
**Pull request by david-german-tri**
@RussTedrake, as requested.

@jwnimmer-tri @sherm1 @psiorx  FYI

### CODE REVIEW

File: drake/doc/developers.rst [hunk 1]
```diff
@@ -8,32 +8,75 @@ The standard github workflow is to fork the drake repository into your
...
+
+The following github users are Drake owners. If possible, seek platform review 
+from an owner who has previously reviewed related changes. Shared context will 
+make the review faster! @RussTedrake is the creator of Drake, but has limited
+time for code review, so prefer other owners unless you really need his input.
+
+- @david-german-tri (Toyota Research Institute)
+- @jwnimmer-tri (Toyota Research Institute)
+- @psiorx (MIT)
+- @sherm1 (Stanford / Toyota Research Institute)
```
Comment (sherm1) [inline drake/doc/developers.rst]: I'm not formally affiliated with Stanford anymore.

**Comment by sherm1** [comment]
Looks great, David -- thanks.

File: drake/doc/developers.rst [hunk 2]
```diff
@@ -8,32 +8,75 @@ The standard github workflow is to fork the drake repository into your
...
-
-Your code will undergo a code review and will run the unit and
-regression tests on a number of build servers.  You can run those
-tests locally by running ``make test`` from the command line -- note
-that there are a lot of computationally demanding tests and this could
-run for a few hours depending on your machine.
-
-Your change must include unit tests that protect it against regressions,
-and those tests must pass on all platforms supported by Drake.  Please
-use the googletest framework, which is already available in the superbuild.
```
Comment (RussTedrake) [inline drake/doc/developers.rst]: c++ code should use googletest.  in matlab, every example and static class method runs as a unit test and fails if it times out or calls matlab's `error` method.  you can direct users to read the doc at `drake/util/unitTest.m` for details on `%NOTEST` and `%TIMEOUT`, etc.

File: drake/doc/developers.rst (hunk 1 above)
Comment (david-german-tri) [inline drake/doc/developers.rst]: Done.

**Comment by Lucy-tri** [comment]
As a feature reviewer of c++ code, I would like a checklist of things to review, both at the high level (I should review that the feature works and is complete), and at the lower, more detail, level.  I'd like a pointer to the current code style guidelines (e.g. Google style for c++, etc.). But this is a good start.

**Comment by liangfok** [comment]
@Lucy-tri: [Here](http://drake002.csail.mit.edu/drake/sphinx/code_style_guide.html#c-style) is the current c++ style guide used by Drake and [here](http://drake002.csail.mit.edu/drake/sphinx/code_style_tools.html#c-c) are some tools for conformance.

**Comment by Lucy-tri** [comment]
@liangfok : thanks, but I was that those pointers would go in developers.rst (or in some doc).

**Comment by liangfok** [comment]
@Lucy-tri, here are the corresponding .rst files. The webpages I linked in my previous post were generated from the .rst files below:

- https://github.com/RobotLocomotion/drake/blob/master/drake/doc/code_style_guide.rst
- https://github.com/RobotLocomotion/drake/blob/master/drake/doc/code_style_tools.rst

**Comment by david-german-tri** [comment]
@RussTedrake If this looks good to you, I think it's ready to merge.

File: drake/doc/developers.rst [hunk 3]
```diff
@@ -8,31 +8,78 @@ The standard github workflow is to fork the drake repository into your
...
+in platform review.
+
+**Platform Review.** After your feature reviewer has signed off on your change,
+reassign it to a Drake owner for platform review. The owner will inspect for
+architectural compatibility, stability, performance, test coverage, and style.
+Once satisfied, the owner merges the PR. 
+
+The following github users are Drake owners. If possible, seek platform review 
+from an owner who has previously reviewed related changes. Shared context will 
+make the review faster! @RussTedrake is the creator of Drake, but has limited
```
Comment (RussTedrake) [inline drake/doc/developers.rst]: while i appreciate the concern, i'd like to remove the line about how busy i am.  :)

Comment (david-german-tri) [inline drake/doc/developers.rst]: Done, sorry!

**Comment by liangfok** [comment]
Just to clarify, after passing the feature reviewer, platform reviewer, and CI tests, can the owner of the PR ultimately hit the "merge pull request" button?

This often happens when the platform reviewer says OK but the CI tests take several more hours to complete.

**Comment by RussTedrake** [comment]
For now, the procedure is that only the short list can merge.  Let’s see how that works.  The build servers will be much faster and more reliable soon.

**Comment by david-german-tri** [comment]
@liangfok I'm hoping that better review tools will allow us to remove that roundtrip, which has definitely been a pain point for me also.  Stay tuned!

**Pull request merged**